src/bp/core/server.ts

```diff
@@ -214,6 +214,9 @@ export default class HTTPServer {
 
     this._needPermissions = needPermissions(this.workspaceService)
     this._hasPermissions = hasPermissions(this.workspaceService)
+
+    // Necessary to prevent circular dependency
+    this.authService.jobService = this.jobService
```

Comment:
Circular dependency when redis is enabled, because of the link between httpServer -> authService -> jobService

Reply:
That is just a very bad smell, why not do it the proper way and refactor the code so we don't have that circular dep around. 

I know we want to go forward and slipping in a few tech dept here and there won't kill us. But we've been doing this for quite too long now. The team will grow and if we want to keep going forward, we'll have to clean this mess one day or another.

Reply:
Moreover, I just looked it up, there is no circular dep between http server auth service and jobService themselves it resides in licensing service

licensingService -> authService -> jobService
jobService -> licensingService

Reply:
Yeah you are right, it was the licensing service. There's a couple of places where we go around the circular dependency in the `botpress.ts` file, but I didn't want to add the job service there because it's not already being used. Still thinking about a better way to fix this